amethyst_tiles/src/iters.rs

```diff
@@ -0,0 +1,215 @@
+//! Various helper iterators for efficiently iterating in coordinate encoded spaces in a locality-friendly fashion.
+
+use crate::morton;
+use amethyst_core::math::Point3;
+
+/// Cubic region stored and handled as a Morton value.
+#[derive(Copy, Clone, Default, Debug, Eq, PartialEq, Hash)]
+pub struct MortonRegion {
+    min: u32,
+    max: u32,
+}
+impl MortonRegion {
+    /// Create a new `MortonRegion` iterator.
+    pub fn new(min: u32, max: u32) -> Self {
+        Self { min, max }
+    }
+
+    /// Check if this `MortonRegion` contains a given morton coordinate.
+    #[inline]
+    pub fn contains(self, morton: u32) -> bool {
+        let m1 = morton::min(self.min, morton) != morton;
+        let m2 = morton::max(self.max, morton) != morton;
```

Comment:
I might misunderstand what's going on there, but for me it seems like the logic checks only the edge conditions, without handling the "inside" of the region.

Reply:
How so? This checks that the morton values are greater than the min, and less than the max.

Reply:
Not really (I think). The test is a bit fuzzy on boundaries. Imagine (or write a test for) what happens when only one of the coordinates is actually different. Also, do you want the range to be inclusive on one side, both, or neither? Because currently it's kinda weird about it. I think it's sort of inclusive on some axes, but not really on others. I think all region corners should also be tested for being inclusive/exclusive.

I have a bit hard time thinking about this in combination with the comparators. Writing it as manually decoding values directly in that method might make it much cleaner.

Reply:
You're correct actually. Per tests, my method was incorrect. 

I've fallen back on decoding and comparing for ease of reading. It could be optimized but its minimal.